From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Tue, 31 Jul 2018 04:44:12 +0100
Subject: [PATCH 13/26] notifications: Handle .desktop files that got renamed

upstream, those use NoDisplay but the panel doesn't respect that key,
skip known buggy entries for file-roller and nautilus
Origin: ubuntu
Bug-Ubuntu: https://launchpad.net/bugs/1716267
Forwarded: not-needed
---
 panels/notifications/cc-notifications-panel.c | 8 ++++++++
 1 file changed, 8 insertions(+)

Index: gnome-control-center/panels/notifications/cc-notifications-panel.c
===================================================================
--- gnome-control-center.orig/panels/notifications/cc-notifications-panel.c
+++ gnome-control-center/panels/notifications/cc-notifications-panel.c
@@ -447,6 +447,14 @@ process_app_info (CcNotificationsPanel *
   guint i;
 
   app_id = app_info_get_id (app_info);
+
+  /* Ignore compatibility desktops (lp: #1716267) */
+  if (g_strcmp0 (app_id, "file-roller") == 0)
+    return;
+
+  if (g_strcmp0 (app_id, "nautilus") == 0)
+    return;
+
   g_strcanon (app_id,
               "0123456789"
               "abcdefghijklmnopqrstuvwxyz"
